lib/experimental.ts

```diff
@@ -1,15 +1,27 @@
-import { PluginResponse } from "./types";
+import * as staticModule from "./static";
+import { ImageType, PluginResponse } from "./types";
 
 export async function experimentalAnalysis(
-  options: any,
+  targetImage: string,
 ): Promise<PluginResponse> {
   // assume Distroless scanning
-  return distroless(options);
+  return distroless(targetImage);
 }
 
 // experimental flow expected to be merged with the static analysis when ready
-export async function distroless(options: any): Promise<PluginResponse> {
+export async function distroless(targetImage: string): Promise<PluginResponse> {
   // assumption #1: the image is present in the local Docker daemon
   // assumption #2: the `docker` binary is available locally
-  throw new Error("not implemented");
+  // TODO: docker-save the image from the daemon to an archive
+  // TODO: generate options for static analysis based on that
+
+  const scanningOptions = {
+    staticAnalysisOptions: {
+      imagePath: "./egg.tar", // TODO
+      imageType: ImageType.DockerArchive,
+      tmpDirPath: "", // TODO only for RPM though?
```

Comment:
Correct! This will soon be removed, because we can feed the RPM file directly into our `rpm-parser` instead of extracting it to file so that some tool can read it from the filesystem 🙏

Reply:
:fanmob: